github.com/cockroachdb/pebble/internal/metamorphic: TestMeta failed

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/14501937?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/14501937?buildTab=artifacts#meta) on refs/heads/master @ [e5c9f635e505](https://github.com/cockroachdb/pebble/commits/e5c9f635e505):


```
        // INFO: [JOB 661] flushing: sstable created 000571
        // INFO: [JOB 661] MANIFEST created 000572
        // INFO: [JOB 661] flushed 1 memtable (173B) to L0 [000571] (796B), in 0.0s (0.0s total), output rate 1.1MB/s
        // INFO: [JOB 661] MANIFEST deleted 000547
        // INFO: [JOB 663] compacting(default) L0 [000560 000571] (1.5KB) Score=100.00 + L5 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // INFO: [JOB 663] compacting: sstable created 000573
        // INFO: [JOB 663] compacted(default) L0 [000560 000571] (1.5KB) Score=100.00 + L5 [] (0B) Score=0.00 -> L5 [000573] (893B), in 0.0s (0.0s total), output rate 90KB/s
        // INFO: [JOB 665] compacting(default) L5 [000573 000568] (1.6KB) Score=24.63 + L6 [000567] (1.1KB) Score=0.26; OverlappingRatio: Single 0.68, Multi 0.00
        // INFO: [JOB 663] sstable deleted 000571
        // INFO: [JOB 665] compacting: sstable created 000574
        // INFO: [JOB 665] compacted(default) L5 [000573 000568] (1.6KB) Score=24.63 + L6 [000567] (1.1KB) Score=0.26 -> L6 [000574] (1.1KB), in 0.0s (0.0s total), output rate 453KB/s
        // INFO: [JOB 666] compacting(default) L0 [000570 000569] (1.5KB) Score=100.00 + L5 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // INFO: [JOB 665] sstable deleted 000567
        // INFO: [JOB 660] ingested L0:000570 (776B), L0:000569 (776B), L5:000568 (776B)
        db1.IngestExternalFiles(external0, "utlyzfoxv" /* start */, "utlyzfoxvduv" /* end */, "" /* syntheticSuffix */, "" /* syntheticPrefix */, external0, "pfnjutlyzfoxv" /* start */, "pfnjutlyzfoxvkmkydajc" /* end */, "" /* syntheticSuffix */, "pfnj" /* syntheticPrefix */, external0, "cautlyzfoxv" /* start */, "cautlyzfoxvbx" /* end */, "@148" /* syntheticSuffix */, "ca" /* syntheticPrefix */) // <nil> #2443
        // INFO: [JOB 666] compacting: sstable created 000575
        db1.Set("bdqj@38", "glfqfvpse") // <nil> #2444
        // INFO: [JOB 665] sstable deleted 000573
        // INFO: [JOB 666] compacted(default) L0 [000570 000569] (1.5KB) Score=100.00 + L5 [] (0B) Score=0.00 -> L5 [000575] (777B), in 0.0s (0.0s total), output rate 413KB/s
        db1.Get("bdqj@29") // [""] pebble: not found #2445
        db1.IngestExternalFiles(external0, "utlyzfoxvfyzrtgrw" /* start */, "utlyzfoxvlaj" /* end */, "@149" /* syntheticSuffix */, "" /* syntheticPrefix */) // no-op #2450
        // INFO: [JOB 669] compacting(default) L5 [000575] (777B) Score=10.43 + L6 [000574] (1.1KB) Score=0.61; OverlappingRatio: Single 1.49, Multi 0.00
        // INFO: [JOB 669] compacting: sstable created 000578
        // INFO: [JOB 668] ingesting: sstable created 000577
        // INFO: [JOB 669] compacted(default) L5 [000575] (777B) Score=10.43 + L6 [000574] (1.1KB) Score=0.61 -> L6 [000578] (1.1KB), in 0.0s (0.0s total), output rate 165KB/s
        // INFO: [JOB 669] sstable deleted 000574
        // INFO: [JOB 671] compacting(default) L5 [000576] (776B) Score=7.00 + L6 [000578] (1.1KB) Score=0.43; OverlappingRatio: Single 1.49, Multi 0.00
        // INFO: [JOB 669] sstable deleted 000575
        // INFO: [JOB 668] ingested L0:000577 (776B), L5:000576 (776B)
        db1.IngestExternalFiles(external0, "utlyzfoxv" /* start */, "utlyzfoxvex" /* end */, "" /* syntheticSuffix */, "" /* syntheticPrefix */, external0, "kimutlyzfoxv" /* start */, "kimutlyzfoxvet" /* end */, "" /* syntheticSuffix */, "kim" /* syntheticPrefix */) // <nil> #2451
        // INFO: [JOB 671] compacting: sstable created 000579
        db1.Merge("szle@78", "namqusdtlhgwp") // <nil> #2452
        db1.Set("ozfb@139", "xygmodpnsyvct") // <nil> #2453
        db1.Get("yconcminh@107") // [""] pebble: not found #2455
        db1.DeleteRange("kllhmbsxunem@21", "tiiirmn@74") // <nil> #2456
        db1.Merge("hmmwvdxe@83", "") // <nil> #2457
        iter19.Close() // <nil> #2458
        // INFO: [JOB 672] sstable deleted 000551
        // INFO: [JOB 671] MANIFEST created 000580
        // INFO: [JOB 671] compacted(default) L5 [000576] (776B) Score=7.00 + L6 [000578] (1.1KB) Score=0.43 -> L6 [000579] (1.1KB), in 0.0s (0.0s total), output rate 505KB/s
        // INFO: [JOB 673] ingested L5:000581 (776B)
        db1.IngestExternalFiles(external0, "utlyzfoxv" /* start */, "utlyzfoxvd" /* end */, "@151" /* syntheticSuffix */, "" /* syntheticPrefix */, external0, "txvkoutlyzfoxvfyzrtgrw" /* start */, "txvkoutlyzfoxvmugd" /* end */, "@150" /* syntheticSuffix */, "txvko" /* syntheticPrefix */) // <nil> #2463
        db1.SingleDelete("gltzry@55", true /* maybeReplaceDelete */) // <nil> #2464
        // INFO: [JOB 671] sstable deleted 000578
        // INFO: [JOB 671] MANIFEST deleted 000562
        iter18.Close() // <nil> #2465
        // INFO: [JOB 677] compacting(default) L5 [000581] (776B) Score=7.00 + L6 [000579] (1.1KB) Score=0.43; OverlappingRatio: Single 1.49, Multi 0.00
        
        To reduce:  go test ./internal/metamorphic -tags invariants -run 'TestMeta$' --run-dir /artifacts/meta/240322-043500.583754581672/random-009 --try-to-reduce -v
        --- FAIL: TestMeta/execution/random-009 (4.82s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run 'TestMeta$' ./internal/metamorphic -seed 1711082100583875351 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/objstorage/objstorageprovider/remote.go
+++ b/objstorage/objstorageprovider/remote.go
@@ -421,5 +421,5 @@ func (p *provider) ensureStorage(locator remote.Locator) (remote.Storage, error)
 func (p *provider) GetExternalObjects(locator remote.Locator, objName string) []base.DiskFileNum {
 	p.mu.Lock()
 	defer p.mu.Unlock()
-	return p.mu.remote.getExternalObjects(locator, objName)
+	return slices.Clone(p.mu.remote.getExternalObjects(locator, objName))
 }

